cli/pkg/gitutil/gitutil.go

```diff
@@ -239,6 +239,19 @@ func CommitAndForcePush(ctx context.Context, projectPath string, config *Config,
 		}
 	}
 
+	// check current branch matches deployed branch
+	headRef, err := repo.Head()
+	if err != nil {
+		return err
+	}
+	if !headRef.Name().IsBranch() {
+		return fmt.Errorf("detached HEAD state detected. Checkout a branch")
+	}
+	branch := headRef.Name().Short()
+	if headRef.Name().Short() != config.DefaultBranch {
+		return fmt.Errorf("current branch %q does not match deployed branch %q", branch, config.DefaultBranch)
```

Comment:
Hmmm these errors would show up in UI as well right? We should have some UI components to either direct user on how to switch or have a button in UI for this.